README.md

```diff
@@ -8,84 +9,98 @@
 [![Puppet Forge - scores](https://img.shields.io/puppetforge/f/puppet/caddy.svg)](https://forge.puppetlabs.com/puppet/caddy)
 [![License](https://img.shields.io/github/license/voxpupuli/puppet-caddy.svg)](https://github.com/voxpupuli/puppet-caddy/blob/master/LICENSE)
 
-#### Table of Contents
+## Table of Contents
 
-1. [Overview](#overview)
-2. [Module Description](#module-description)
-3. [Usage](#usage)
-4. [Limitations](#limitations)
-5. [TODO](#TODO)
+1. [Description](#description)
+1. [Setup - The basics of getting started with Caddy](#setup)
+    * [What Caddy affects](#what-Caddy-affects)
+    * [Setup requirements](#setup-requirements)
+    * [Beginning with Caddy](#beginning-with-Caddy)
+1. [Usage - Configuration options and additional functionality](#usage)
+1. [Limitations - OS compatibility, etc.](#limitations)
+1. [Development - Guide for contributing to the module](#development)
 
-## Overview
+## Description
 
-Puppet Caddy module installs and configures caddy - The HTTP/2 web server with automatic HTTPS.
+This module installs and configures Caddy - The HTTP/2 web server with automatic
+HTTPS.
 
-## Module Description
+## Setup
 
-Puppet Caddy module handles installing, configuring, and running Caddy server on Redhat based operating systems.
+### What Caddy affects
 
-## Usage
+* Caddy binary
+* Caddy configuration file
+* Caddy virtual hosts
+* Caddy service
+
+### Setup Requirements
+
+This module has the following dependencies:
```

Comment:
Do we need to list these? IMHO metadata.json is a better place.

Reply:
I have no idea. We have these dependencies at ```metadata.json```, but it is nice to get an quick overview without jumping through all these files. 

However, it is almost impossible to create a good ```README.md```, because all of our modules use a different style and we do not have a reference module like ```puppetlabs/puppetlabs-apt```.